gpytorch/kernels/keops/matern_kernel.py

```diff
@@ -43,8 +44,17 @@ def covar_func(self, x1, x2, diag=False):
                 return constant_component * exp_component
             else:
                 with torch.autograd.enable_grad():
-                    x1_ = KEOLazyTensor(x1[..., :, None, :])
-                    x2_ = KEOLazyTensor(x2[..., None, :, :])
+                    # We only should use KeOps on big kernel matrices
+                    # If we would otherwise be performing Cholesky inference, then don't apply KeOps
+                    if (
+                        x1.size(-2) < settings.max_cholesky_size.value()
+                        or x2.size(-2) < settings.max_cholesky_size.value()
+                    ):
+                        x1_ = x1[..., :, None, :]
```

Comment:
I know that N is gonna be inherently smallish here, but I think we should avoid materializing an n x n x d tensor here when we get to the distance calculation below just because it's easy enough to avoid. Let's move the distance calculation inside and use covar_dist / the RBF Function as we do in RBFKernel.

Reply:
Cool. Updated the files, and checked that they both work in the CIQ sampling example.